lib/codegen/selection/generator.cc

```diff
@@ -982,6 +982,19 @@ void generator::visit_sin_inst(ir::sin_inst* x){
   }
  }
 
+/**
+ * \brief Code Generation for `tanh`
+ */
+void generator::visit_tanh_inst(ir::tanh_inst* x){
+  std::vector<llvm::Type*> tys = {f32_ty};
+  FunctionType *fn_ty = FunctionType::get(f32_ty, tys, false);
+  InlineAsm *tanh = InlineAsm::get(fn_ty, "tanh.approx.f32 $0, $0;", "=f,0", false);
```

Comment:
`tanh.approx.f32` is (IIUC) sm_75+ (Turing & Ampere). How would we want to handle that?

Reply:
@madeleineth good point! @nelhage is it something that you guys actively use? If yes, I think it's fine to throw an exception when `sm < 70` is detected. You can use https://github.com/openai/triton/blob/master/include/triton/codegen/selection/generator.h#L209 for that. Of course, if we go this route, the unit tests would have to be adjusted to make sure that the right exception is thrown on Volta.

FWIW, I think the long term solution here is probably to have wrappers for nvidia's `libdevice`. This should be sort-of easy to do once forward-declaration and external linkage of element-wise functions is supported. Most of the hard work for this is done, but not merged, in the `device-functions` branch.

Reply:
We're not actively using it yet, but we were looking at fusing gelu with some other ops, using the `tanh` approximation thereof. I can take a stab at adding that check in a bit.